contracts/axelar-gas-service/src/test.rs

```diff
@@ -22,14 +26,87 @@ fn setup_env<'a>() -> (Env, Address, Address, AxelarGasServiceClient<'a>) {
 
     let client = AxelarGasServiceClient::new(&env, &contract_id);
     let gas_collector: Address = Address::generate(&env);
-    client.initialize(&gas_collector);
 
     (env, contract_id, gas_collector, client)
 }
 
+#[test]
+fn test_initialize() {
+    let (env, contract_id, gas_collector, client) = setup_env();
+    client.initialize(&gas_collector);
+
+    assert!(env.as_contract(&contract_id, || {
+        env.storage()
+            .instance()
+            .get::<DataKey, bool>(&DataKey::Initialized)
+            .unwrap()
+    }));
+
+    assert_eq!(
+        env.as_contract(&contract_id, || {
+            env.storage()
+                .instance()
+                .get::<DataKey, Address>(&DataKey::GasCollector)
+                .unwrap()
+        }),
+        gas_collector
+    );
+}
+
+#[test]
+fn fail_already_initialized() {
+    let (_env, _contract_id, gas_collector, client) = setup_env();
+    client.initialize(&gas_collector);
+
+    assert_contract_err!(
+        client.try_initialize(&gas_collector),
+        GasServiceError::AlreadyInitialized
+    );
+}
+
+#[test]
+fn fail_pay_gas_invalid_amount() {
+    let (env, contract_id, gas_collector, client) = setup_env();
+
+    let token_address: Address = env.register_stellar_asset_contract(Address::generate(&env));
```

Comment:
```suggestion
    let asset = StellarAssetClient::new (env.register_stellar_asset_contract(Address::generate(&env)));
```
then you can reuse it later, and get the address via `asset.address`

Reply:
done in [dc05398](https://github.com/axelarnetwork/axelar-cgp-soroban/pull/26/commits/dc05398e7b4a479295fa4a497815d0c2ba163fba).  the function `register_stellar_asset_contract_v2` returns a `StellarAssetClient` and `register_stellar_asset_contract` is deprecated in favour of `v2`, so I used that instead for these changes.